control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -0,0 +1,270 @@
+		coreconfig.IncrementContractGeneration(ct)
+
+		// Update the configuration map with the new contract data.
+		if err := r.UpdateDataPlaneConfigMap(ctx, ct, contractConfigMap); err != nil {
+			logger.Error("failed to update data plane config map", zap.Error(
+				statusConditionManager.FailedToUpdateConfigMap(err),
+			))
+			return err
+		}
+		logger.Debug("Contract config map updated")
+	}
+	statusConditionManager.ConfigMapUpdated()
+
+	// We update receiver and dispatcher pods annotation regardless of our contract changed or not due to the fact
+	// that in a previous reconciliation we might have failed to update one of our data plane pod annotation, so we want
+	// to anyway update remaining annotations with the contract generation that was saved in the CM.
+
+	// We reject events to a non-existing Channel, which means that we cannot consider a Channel Ready if all
+	// receivers haven't got the Channel, so update failures to receiver pods is a hard failure.
+	// On the other side, dispatcher pods care about Subscriptions, and the Channel object is used as a configuration
+	// prototype for all associated Subscriptions, so we consider that it's fine on the dispatcher side to receive eventually
+	// the update even if here eventually means seconds or minutes after the actual update.
+
+	// Update volume generation annotation of receiver pods
+	if err := r.UpdateReceiverPodsAnnotation(ctx, logger, ct.Generation); err != nil {
+		logger.Error("Failed to update receiver pod annotation", zap.Error(
+			statusConditionManager.FailedToUpdateReceiverPodsAnnotation(err),
+		))
+		return err
+	}
+	logger.Debug("Updated receiver pod annotation")
+
+	// Update volume generation annotation of dispatcher pods
+	if err := r.UpdateDispatcherPodsAnnotation(ctx, logger, ct.Generation); err != nil {
+		// Failing to update dispatcher pods annotation leads to config map refresh delayed by several seconds.
+		// Since the dispatcher side is the consumer side, we don't lose availability, and we can consider the Channel
+		// ready. So, log out the error and move on to the next step.
+		logger.Warn(
+			"Failed to update dispatcher pod annotation to trigger an immediate config map refresh",
+			zap.Error(err),
+		)
+
+		statusConditionManager.FailedToUpdateDispatcherPodsAnnotation(err)
+	} else {
+		logger.Debug("Updated dispatcher pod annotation")
+	}
+
+	return statusConditionManager.Reconciled()
+}
+
+//nolint
+func (r *Reconciler) FinalizeKind(ctx context.Context, channel *messagingv1beta1.KafkaChannel) reconciler.Event {
+	// TODO: noop for now
+	return nil
+}
+
+//nolint
+func (r *Reconciler) topicConfig() (*kafka.TopicConfig, error) {
+	// TODO: all hardcoded for now
+	return &kafka.TopicConfig{
+		TopicDetail: sarama.TopicDetail{
+			NumPartitions:     1,
+			ReplicationFactor: 1,
+			ReplicaAssignment: nil,
+			ConfigEntries:     nil,
+		},
+		BootstrapServers: []string{"my-cluster-kafka-bootstrap.kafka:9092"},
+	}, nil
+}
```

Comment:
I've already check to factor this into some more common package - want me to help on that?  and you move forward w/ the actual channel?